test/distributed/org/apache/cassandra/distributed/test/guardrails/GuardrailNonPartitionRestrictedQueryTest.java

```diff
@@ -0,0 +1,450 @@
+                cluster.get(nodesToFlush).forEach((instance) -> instance.flush(KEYSPACE));
+        }
+
+        return value;
+    }
+
+    private void setThresholds(int warn, int fail, int... nodes)
+    {
+        Stream<IInvokableInstance> instances;
+
+        if (nodes.length == 0)
+            instances = cluster.stream();
+        else
+            instances = cluster.get(nodes).stream();
+
+        instances.forEach(instance -> instance.acceptsOnInstance((IIsolatedExecutor.SerializableBiConsumer<Integer, Integer>)
+                                                                 (w, f) -> Guardrails.instance.setSaiSSTableIndexesPerQueryThreshold(w, f))
+                                              .accept(warn, fail));
+        assertTresholds(warn, fail, nodes);
+    }
+
+    private void enableGuardrail()
+    {
+        cluster.forEach(instance -> instance.runOnInstance((IIsolatedExecutor.SerializableRunnable) () -> Guardrails.instance.setNonPartitionRestrictedQueryEnabled(true)));
+        cluster.forEach(instance -> assertTrue(instance.callsOnInstance((IIsolatedExecutor.SerializableCallable<Boolean>) () -> Guardrails.instance.getNonPartitionRestrictedIndexQueryEnabled()).call()));
+    }
+
+    private void disableGuardrail()
+    {
+        cluster.forEach(instance -> instance.runOnInstance((IIsolatedExecutor.SerializableRunnable) () -> Guardrails.instance.setNonPartitionRestrictedQueryEnabled(false)));
+        cluster.forEach(instance -> assertFalse(instance.callsOnInstance((IIsolatedExecutor.SerializableCallable<Boolean>) () -> Guardrails.instance.getNonPartitionRestrictedIndexQueryEnabled()).call()));
+    }
+
+    private void assertTresholds(int expectedWarn, int expectedFail, int... nodes)
+    {
+        Stream<IInvokableInstance> instances;
+
+        if (nodes.length == 0)
+            instances = cluster.stream();
+        else
+            instances = cluster.get(nodes).stream();
+
+        instances.forEach(instance -> {
+            assertEquals(expectedWarn,
+                         instance.callsOnInstance((IIsolatedExecutor.SerializableCallable<Integer>) () -> Guardrails.instance.getSaiSSTableIndexesPerQueryWarnThreshold())
+                                 .call().intValue());
+            assertEquals(expectedFail,
+                         instance.callsOnInstance((IIsolatedExecutor.SerializableCallable<Integer>) () -> Guardrails.instance.getSaiSSTableIndexesPerQueryFailThreshold())
+                                 .call().intValue());
+        });
+    }
+
+    private long totalWarnings()
+    {
+        return cluster.stream().mapToLong(i -> i.metrics().getCounter("org.apache.cassandra.metrics.keyspace.TooManyIndexReadSSTablesWarnings.distributed_test_keyspace")).sum();
+    }
+
+    private long totalAborts()
+    {
+        return cluster.stream().mapToLong(i -> i.metrics().getCounter("org.apache.cassandra.metrics.keyspace.TooManyIndexReadSSTablesAborts.distributed_test_keyspace")).sum();
```

Comment:
nit: Alternate, more consistent naming
```suggestion
        return cluster.stream().mapToLong(i -> i.metrics().getCounter("org.apache.cassandra.metrics.keyspace.TooManySSTableIndexesReadAborts.distributed_test_keyspace")).sum();
```